mesonpy/_tags.py

```diff
@@ -1,130 +1,124 @@
+    'cpython': 'cp',
+    'pypy': 'pp',
+    'ironpython': 'ip',
+    'jython': 'jy',
+}
+
+
+def get_interpreter_tag() -> str:
+    name = sys.implementation.name
+    name = INTERPRETERS.get(name, name)
+    version = sys.version_info
+    return f'{name}{version[0]}{version[1]}'
+
+
+def _get_config_var(name: str, default: Union[str, int, None] = None) -> Union[str, int, None]:
+    value = sysconfig.get_config_var(name)
+    if value is None:
+        return default
+    return value
+
+
+def _get_cpython_abi() -> str:
+    version = sys.version_info
+    debug = pymalloc = ''
+    if _get_config_var('Py_DEBUG', hasattr(sys, 'gettotalrefcount')):
+        debug = 'd'
+    if sys.version_info < (3, 8) and _get_config_var('WITH_PYMALLOC', True):
+        pymalloc = 'm'
+    return f'cp{version[0]}{version[1]}{debug}{pymalloc}'
+
+
+def get_abi_tag() -> str:
+    abi = sysconfig.get_config_var('SOABI')
+    if not abi:
+        # CPython on Windows does not have a SOABI sysconfig variable.
+        assert sys.implementation.name == 'cpython'
+        return _get_cpython_abi()
+    if abi.startswith('cpython'):
+        return 'cp' + abi.split('-')[1]
+    if abi.startswith('pypy'):
+        return '_'.join(abi.split('-')[:2])
+    return abi.replace('.', '_').replace('-', '_')
+
+
+def _get_macosx_platform_tag() -> str:
+    ver, x, arch = platform.mac_ver()
+
+    # Override the macOS version if one is provided via the
+    # MACOS_DEPLOYMENT_TARGET environment variable.
+    try:
+        version = tuple(map(int, os.environ.get('MACOS_DEPLOYMENT_TARGET', '').split('.')))[:2]
+    except ValueError:
+        version = tuple(map(int, ver.split('.')))[:2]
+
+    # Python built with older macOS SDK on macOS 11, reports an
+    # unexising macOS 10.16 version instead of the real version.
+    #
+    # The packaging module introduced a workaround
+    # https://github.com/pypa/packaging/commit/67c4a2820c549070bbfc4bfbf5e2a250075048da
+    #
+    # This results in packaging versions up to 21.3 generating
+    # platform tags like "macosx_10_16_x86_64" and later versions
+    # generating "macosx_11_0_x86_64".  Using latter would be more
+    # correct but prevents the resulting wheel from being installed on
+    # systems using packaging 21.3 or earlier (pip 22.3 or earlier).
+    #
+    # Fortunately packaging versions carrying the workaround still
+    # accepts "macosx_11_0_x86_64" as a compatible platform tag.  We
+    # can therefore ignore the issue and generate the slightly
+    # incorrect tag.
+
+    major, minor = version
+
+    if major >= 11:
+        # For macOS reelases up to 10.15, the major version number is
+        # actually part of the OS name and the minor version is the
+        # actual OS release.  Starting with macOS 11, the major
+        # version number is the OS release and the minor version is
+        # the patch level.  Reset the patch level to zero.
+        minor = 0
+
+    if sys.maxsize <= 2**32:
```

Comment:
nitpick: Make this a constant.

Reply:
The only name for the constant I can think about is `_2_POWER_32`, which is maybe more cryptic than the literal value. Do you have a specific name that would be clearer than the value itself?

Reply:
`sys.maxsize < 2 ** 32` would be equivalent to `sys.maxsize <= UINT32_MAX`; since this isn't Windows, that's likely the same as `sys.maxsize <= ULONG_MAX`.  I suspect this is being used as a delineation between 32-bit Python and 64-bit Python, so you could also choose a name based on that (`PYTHON_32BIT_SYS_MAXSIZE_CUTOFF`, maybe)

Reply:
Maybe the suggestion was to use `if sys.maxsize < 4294967296` but I'm not sure whether that's all that much of an advantage...

Reply:
I mean, make `sys.maxsize <= 2**32` a constant, not `2**3`. Something like `_32_BIT_INTERPRETER`, just like `packaging` does.

Reply:
I'll do that but I really don't see the advantage.